### DISCUSSION TITLE
v2.3.1

### DISCUSSION THREAD

**Discussion by alexdelprete** [alexdelprete/ha-abb-powerone-pvi-sunspec#47] [Announcements]
- Fixes ConfigFlow issue #46 

<hr /><em>This discussion was created from the release <a href='https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/releases/tag/v2.3.1'>v2.3.1</a>.</em>

**Comment by Assiolo** [comment]
error in configuration spet by click on new integration -> ABB Power-One PVI SunSpec
Errore
Impossibile caricare il flusso di configurazione: {"message":"Invalid handler specified"}

**Reply by alexdelprete** [reply]
Altre persone l'hanno testata, e funziona: https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/issues/46#issuecomment-1377004714

Hai riavviato HA prima di provare?

**Reply by alexdelprete** [reply]
Se vuoi, passa direttamente a questa versione, così fai un test sull'ultima: https://community.home-assistant.io/t/custom-component-abb-power-one-fimer-pv-inverters-sunspec-modbus-tcp/316363/224?u=alexdelprete

**Reply by Assiolo** [reply]
si ho riavviato anche la macchina stessa, una volta riavviato mi dice avvio configurazione e poi mi dà l'errore

**Reply by Assiolo** [reply]
ho provato anche con la release candidate 1 ma stesso problema non so perchè qualche mese fa l'aveva installata con successo, poi causa batterie nel sistema ho provato sunspec, ora volendo reinstallare power one mi da questo errore, non c'è modo di intallarlo tramite yaml? come workaround?

**Reply by alexdelprete** [reply]
disinstallala dalle integrazioni, e disinstallala anche da HACS, riavvia HA, poi vai nel folder custom_components e accertati che non ci sia più il folder `abb_powerone_pvi_sunspec`.

a quel punto dovresti aver pulito i residui...poi prova a reinstallare da HACS, riavvia, poi aggiungi l'integrazione.

**Reply by Assiolo** [reply]
si ho fatto così ma niente continua a darmi errore ed è l'unico modo per installarla...

**Reply by alexdelprete** [reply]
se hai la possibilità di installare un HA su un server di test, un istanza pulita, prova ad installare HACS e il componente, giusto per verificare ci sia qualcosa nel tuo ambiente che stia creando il problema.

qui funziona tutto regolarmente, e anche altri utenti lo hanno installato senza riportare problemi.

se usi la RC1, ultima versione, devi avere almeno HA 2023.1.

**Reply by Assiolo** [reply]
io ho HA aggiornato all'ultima versione Home
Assistant 2023.1.2
Supervisor 2022.12.1
Operating System 9.4
Frontend 20230104.0 - latest
attualmente avevo/ho (perchè ho reintegrato iol file yaml modbus) installato sunspec
powerone lo avevo 2 versioni fa di ha, purtroppo non ho modo di verificare una nuova installazione di ha

**Reply by alexdelprete** [reply]
Non puoi utilizzare il mio componente, quello Sunspec e il modbus yaml contemporaneamente. Solo uno dei 3, perché altrimenti vanno in conflitto accedendo allo stesso dispositivo.

Devi scegliere uno dei tre e disinstallare gli altri due.

**Comment by Assiolo** [comment]
No aspetta non hai capito i componenti sono 2 sunspec e il tuo, avevo il tuo di un po' di versioni fa, ho disintallato tutto e installato sunspec che implementa lui modbus tramite yaml per identificare le entità, ora con la tua versione nuova ho tolto lo yaml aggiunto, disintallato sunspec e non sono riuscito a reinstallare abb da due giorni, oggi ho rimesso sunspec per bisogno ma il problema si presentava sensa altro installato

**Comment by alexdelprete** [comment]
hai scritto tu che hai reintegrato modbus yaml (sai cosa significa? integrazione nativa modbus di HA via file yaml). poi dici che stai usando sunspec ma che hai provato ad installare il mio ABB. Boh...non si capisce nulla.

il componente sunspec che ti ho suggerito non si configura via yaml, è come il mio, si configura via configflow.

mi sa che fai una confusione incredibile...per quello credo tu abbia problemi...comunque, mi spiace, l'ho testato su HA vergine e il component si installa senza problemi, come hanno confermato anche altri.

quindi, in tutto questo caos, sinceramente non so come aiutarti, sorry.

**Comment by Assiolo** [comment]
guarda uso quel componente da un po adesso, e invece si configura via yaml ahahaha, nel senso che si installa come componente poi tramite yaml vai a dare delle specifiche, io ad esempio gli ho tirato fuori i dati della batteria andando a cambiargli anche i nomi entità, non sto facendo confusione, i problemi li ho per un altro motivo, come vedi se fai delle ricerche è un problema che salta fuori , ma non importa non devi risolvere un problema se davvero è solo mio grazie comunque

**Comment by alexdelprete** [comment]
il componente che ti ho segnalato io è questo: https://github.com/CJNE/ha-sunspec
si configura come il mio, non via yaml...mah, ridi pure, ma continui a non capire...:)

tu stai configurando l'integrazione MODBUS via yaml (cioè questa: https://www.home-assistant.io/integrations/modbus/), quella è integrazione NATIVA HA. C'entra una mazza sunspec. Stai facendo MOLTA confusione, credimi.

**Comment by Assiolo** [comment]
Mah ti credo,  ma se uso lo yaml senza sunspec (lo intallo tramite componenti aggiuntivi) non vede nulla, viceversa se installo solo sunspec senza yaml non edito le entità come ho fatto, detto cio' mi fido provo ad eliminare tutto, tutto e riprovo con la tua, grazie comunque del tuo tempo!

**Comment by alexdelprete** [comment]
Assiolo, modbus: è una INTEGRAZIONE NATIVA, il mio ABB è un componente custom, ha-sunspec è un componente custom. Tutti e tre lavorano via modbus, che è un protocollo di comunicazione. Non si usano INSIEME per lo stesso device. Nessuno di loro ha bisogno dell'altro per funzionare.

Non so cosa tu stia facendo, ma non ti funzionerà mai nulla se smanetti senza capire come funzionino le cose.

Per l'inverter, o usi modbus oppure il mio ABB oppure ha-sunspec. Decidi quale usare. Funzionano tutti e tre, ma se usi solo modbus nativo devi mettere nel file yaml tutti gli indirizzi dei registri che vuoi leggere, poco pratico.

Se li installi e configuri tutti o un paio contemporaneamente, fai solo casini, accedendo allo stesso device da due integrazioni diverse.

**Comment by Assiolo** [comment]
rifatto tutto eccolo il log:
Logger: homeassistant.config_entries
Source: config_entries.py:852
First occurred: 23:18:36 (1 occurrences)
Last logged: 23:18:36

Error occurred loading configuration flow for integration abb_powerone_pvi_sunspec: cannot import name 'ModbusTcpClient' from 'pymodbus.client' (/usr/local/lib/python3.10/site-packages/pymodbus/client/__init__.py)

**Comment by alexdelprete** [comment]
Lui aveva lo stesso errore, e ha risolto così: https://community.home-assistant.io/t/custom-component-abb-power-one-fimer-pv-inverters-sunspec-modbus-tcp/316363/196?u=alexdelprete

Non hai eliminato modbus: (integrazione nativa HA) dalla configurazione.

**Comment by Assiolo** [comment]
ma per eliminarla non basta toglierla dallo yaml?

**Comment by Assiolo** [comment]
pero' aspetta ho in yaml un altro script sempre modbus che mi serve per leggere i dati dal contatore (openmeter) sarà quello? ma quello mi serve

**Comment by alexdelprete** [comment]
commenta quel modbus, non cancellare, installa il component, e quando funziona poi rimetti il modbus openmeter e verifichi se funziona o ritorna a confliggere.

comunque è strano...mai avuto questi problemi...

**Comment by Assiolo** [comment]
lui ha risolto togliendolo ora elimino tiutte le voci modbus da yaml e provo

**Reply by alexdelprete** [reply]
commentare vuol dire togliere, senza cancellare!!! 😄 

metti un cancelletto davanti a tutte le righe... #

**Comment by Assiolo** [comment]
infatti confermo funziona, non devo avere la vode modbus nello yaml (ho preferito toglierlo or installo e poi lo rimetto) tanto è un codice semplice che mi deve leggere openmeter all'indirizzo ip

**Reply by alexdelprete** [reply]
ma che usi Alfa di Sinapsi?

**Reply by Assiolo** [reply]
bravissimo! proprio lui

**Reply by alexdelprete** [reply]
sto preparando il custom component per l'Alfa. :)

**Comment by Assiolo** [comment]
ora ho installato ho messo ip porta ecc, ma sotto la scheda mi rimane con nessun entità e la scritta configura,

**Reply by alexdelprete** [reply]
quale scheda? fammi screenshot

**Reply by Assiolo** [reply]
![Immagine 2023-01-13 001942](https://user-images.githubusercontent.com/117266498/212201844-229e7911-5380-4a53-883f-166c1bb78a1e.jpg)

**Comment by Assiolo** [comment]
ok lo ha trovato alla 254 non 247 e mettendo 40000 (ho react 2)

**Reply by alexdelprete** [reply]
se non configuri bene i parametri...quelli di default mica vanno bene per tutti. :)

ora quali entità vedi?

**Comment by Assiolo** [comment]
ora ne vedo 18,
![Immagine 2023-01-13 002756](https://user-images.githubusercontent.com/117266498/212202763-981104be-63df-4a42-b009-4e10e676e0f7.jpg)

**Reply by Assiolo** [reply]
alla fine sono 9 quelle utili

**Comment by alexdelprete** [comment]
Fammi vedere la schermata con i valori, cliccando sul device (vedi i miei sensori unavailable perché il mio inverter è spento, non ho le batterie di accumulo). Catturami la colonna Sensors completa...voglio vedere tutti i valori.

![image](https://user-images.githubusercontent.com/7027842/212202998-5254f663-9efc-4d15-8fe9-f3613f3df0df.png)

**Reply by Assiolo** [reply]
![Immagine 2023-01-13 003342](https://user-images.githubusercontent.com/117266498/212203453-00529cbb-5c84-4cae-97c5-d3ddb85bf5aa.jpg)

**Comment by alexdelprete** [comment]
Funziona tutto vedo, il modello pure era nella lista di quelli conosciuti. Ottimo.

Ora cosa ti serve? :)

**Reply by Assiolo** [reply]
ora il mio problema rimane il flusso carica scarica e stato batteria, ma soprattutto e che se rimetto la lettura a sinapsi scoppia lo yaml

**Reply by alexdelprete** [reply]
aridaje con questo yaml!! :)

la batteria non l'ho implementata perché non avendola non posso testare nulla. quindi non ho mai studiato quali siano i registri e se effettivamente i registri siano gli stessi per tutti.

per quanto concerne l'alfa: fammi vedere il file di configurazione che usi per il modbus: e quando lo riabiliti, e riavvii HA, che errore hai?

**Reply by Assiolo** [reply]
Component error: modbus - Exception importing homeassistant.components.modbus

**Reply by alexdelprete** [reply]
fammi vedere il file di configurazione

**Reply by Assiolo** [reply]
modbus:
  - name: Alfa
    type: tcp
    host: 192.168.178.95
    port: 502
    sensors:
      - name: Alfa_Potenza_Prelevata_Istantanea
        device_class: power
        unit_of_measurement: W
        state_class: measurement
        data_type: uint16
        address: 2
        count: 1
        unique_id: Alfa_Potenza_Prelevata_Istantanea
      - name: Alfa_Potenza_Immessa_Istantanea
        device_class: power
        unit_of_measurement: W
        state_class: measurement
        data_type: uint16
        address: 12
        count: 1
        unique_id: Alfa_Potenza_Immessa_Istantanea
      - name: Alfa_Potenza_Prodotta_Istantanea
        device_class: power
        unit_of_measurement: W
        state_class: measurement
        data_type: uint16
        address: 921
        count: 1
        unique_id: Alfa_Potenza_Prodotta_Istantanea
      - name: Alfa_Energia_Prelevata_Totale
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 5
        count: 2
        unique_id: Alfa_Energia_Prelevata_Totale
      - name: Alfa_Energia_Immessa_Totale
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 15
        count: 2
        unique_id: Alfa_Energia_Immessa_Totale
      - name: Alfa_Energia_Prodotta_Totale
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 924
        count: 2
        unique_id: Alfa_Energia_Prodotta_Totale
      - name: Pot Att Prel Quart Media
        device_class: power
        unit_of_measurement: W
        state_class: measurement
        data_type: uint16
        address: 9
        count: 1
      - name: Pot Att Imm Quart Media
        device_class: power
        unit_of_measurement: W
        state_class: measurement
        data_type: uint16
        address: 19
        count: 1
      - name: Tot Energ Att Prel Giorno-1 F1
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 30
        count: 2
      - name: Tot Energ Att Prel Giorno-1 F2
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 32
        count: 2
      - name: Tot Energ Att Prel Giorno-1 F3
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 34
        count: 2
      - name: Tot Energ Att Prel Giorno-1 F4
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 36
        count: 2
      - name: Tot Energ Att Prel Giorno-1 F5
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 38
        count: 2
      - name: Tot Energ Att Prel Giorno-1 F6
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 40
        count: 2
      - name: Tot Energ Att Imm Giorno-1 F1
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 54
        count: 2
      - name: Tot Energ Att Imm Giorno-1 F2
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 56
        count: 2
      - name: Tot Energ Att Imm Giorno-1 F3
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 58
        count: 2
      - name: Tot Energ Att Imm Giorno-1 F4
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 60
        count: 2
      - name: Tot Energ Att Imm Giorno-1 F5
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 62
        count: 2
      - name: Tot Energ Att Imm Giorno-1 F6
        device_class: energy
        unit_of_measurement: Wh
        state_class: total_increasing
        data_type: uint32
        address: 64
        count: 2
      - name: Fascia oraria corrente
        data_type: uint16
        address: 203
        count: 1
      - name: Data evento
        data_type: uint32
        address: 780
        count: 2
      - name: Tempo residuo distacco
        data_type: uint16
        address: 782
        count: 1

**Reply by alexdelprete** [reply]
e questo l'hai messo in configuration.yaml?

**Reply by Assiolo** [reply]
si

**Reply by alexdelprete** [reply]
Io all'inizio usavo il mio componente e questo codice (messo in un package file però), e non ho mai avuto quell'errore.

**Reply by alexdelprete** [reply]
ora l'Alfa lo interfaccio via Node-Red (sulla dx sono tutti sensori in HA che Node-Red aggiorna automaticamente):

![image](https://user-images.githubusercontent.com/7027842/212205128-ca267097-10b4-467e-bd25-5efe92943be8.png)

**Reply by Assiolo** [reply]
bè dai sei già impazzito abbastanza amen, volevo usare il tuo pacchetto perchè è molto attivo in aggiornamenti ma ho bbisogno di sinapsi per il monitoraggio completo

**Reply by Assiolo** [reply]
purtroppo non conosco il funzionamento di nodered

**Reply by alexdelprete** [reply]
> bè dai sei già impazzito abbastanza amen, volevo usare il tuo pacchetto perchè è molto attivo in aggiornamenti ma ho bbisogno di sinapsi per il monitoraggio completo

ti ripeto, io usavo entrambi all'inizio, senza problemi. Non so perché ti dia errore sinceramente. Un componente non va in conflitto con un altro, se si usano per device differenti (inverter + alfa).

in ogni caso, se aspetti un po', rilascerò il componente per Alfa, così non dovrai usare più l'integrazione modbus: nativa.

**Reply by Assiolo** [reply]
grande grazie! se ti è utile ti do anche il codice che ho nel configuration yaml modbus nativo magari è utile leggere gli indirizzi per le batterie? magari invece non ti serve a nulla

**Reply by alexdelprete** [reply]
il giorno che deciderò di implementare il supporto batterie te lo chiederò, ho notato che i valori sono diversi per alcuni sistemi, e questo mi lascia dubbioso. se fossero gli stessi registri per tutti, allora potrei implementare, ma se sono diversi diventa difficile...vedremo in seguito...

**Reply by Assiolo** [reply]
;) e grazie ancora di tutto!

**Reply by alexdelprete** [reply]
> ;) e grazie ancora di tutto!

dimenticavo una cosa: hai usato la v2.5.0-rc1 del mio componente? Hai provato la funzione CONFIGURE quando hai modificato i parametri? Vedi video qui della nuova funzione di configurazione dei parametri: https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/releases/tag/v2.5.0-rc.1
